### ISSUE TITLE
Bogus error while assigning via function pointer

### ISSUE THREAD

**Issue by asl** [llvm/llvm-project#7101]
|  |  |
| --- | --- |
| Bugzilla Link | [6729](https://llvm.org/bz6729) |
| Version | trunk |
| OS | Linux |

## Extended Description 
Consider the following testcase:
```
$ cat test.cpp
void func(int x) {
}

typedef void (*f_type)(int x) __attribute__((noreturn));

int main() {
    f_type f = &func;
    f(10);
}
```
I'm seeing:
```
$ ./clang -c test.cpp 
test.cpp:8:12: error: cannot initialize a variable of type 'f_type'
(aka 'void (*)(int) __attribute__((noreturn))') with an
      rvalue of type 'void (*)(int)'
    f_type f = &func;
           ^   ~~~~~
1 diagnostic generated.
```
g++ is silent even in pedantic mode

**Comment by JOE1994** [comment]
`func` doesn't seem to be a `noreturn` function ?

**Comment by shafik** [comment]
[cwg836](https://www.open-std.org/jtc1/sc22/wg21/docs/cwg_closed.html#836) tell us that it is not part of the type but `__attribute__((noreturn))` is different: https://github.com/llvm/llvm-project/issues/113511 but based on the conversation.

It looks like based on that conversation clang should allow the conversion.

CC @erichkeane @AaronBallman

**Comment by erichkeane** [comment]
> [cwg836](https://www.open-std.org/jtc1/sc22/wg21/docs/cwg_closed.html#836) tell us that it is not part of the type but `__attribute__((noreturn))` is different: #113511 but based on the conversation.
> 
> It looks like based on that conversation clang should allow the conversion.
> 
> CC @erichkeane @AaronBallman

Yeah, this seems to be an inverse-duplicate of 113511.  I might suggest closing this one and using that one, since it has the full analysis on it.
